scripts/wrappers/addons.py

```diff
@@ -18,6 +21,123 @@
+        click.echo(f"    microk8s addons repo remove {repo_name}", err=True)
+        sys.exit(1)
+    except yaml.YAMLError as err:
+        click.echo(f"Yaml format error in addons.yaml file: {err}", err=True)
+        sys.exit(1)
+
+    try:
+        validate_yaml_schema(contents)
+    except jsonschema.ValidationError as err:
+        click.echo(f"Invalid addons.yaml file: {err.message}", err=True)
+        sys.exit(1)
+
+
+def load_addons_yaml(repo_dir: Path):
+    addons_yaml = repo_dir / "addons.yaml"
+    with open(addons_yaml, mode="r") as f:
+        return yaml.safe_load(f.read())
+
+
+def validate_yaml_schema(contents):
+    """
+    Validates that the addons.yaml file has the expected format.
+    """
+    schema = {
+        "type": "object",
+        "properties": {
+            "microk8s-addons": {
+                "type": "object",
+                "properties": {
+                    "addons": {
+                        "type": "array",
+                        "items": {
+                            "type": "object",
+                            "properties": {
+                                "name": {"type": "string"},
+                                "description": {"type": "string"},
+                                "version": {"type": "string"},
+                                "check_status": {"type": "string"},
+                                "supported_architectures": {
+                                    "type": "array",
+                                    "items": {
+                                        "type": "string",
+                                        "enum": ["amd64", "arm64", "s390x"],
+                                    },
+                                },
+                            },
+                            "required": [
+                                "name",
+                                "description",
+                                "version",
+                                "check_status",
+                                "supported_architectures",
+                            ],
+                        },
+                    }
+                },
+                "required": ["addons"],
+            }
+        },
+        "required": ["microk8s-addons"],
+    }
+    jsonschema.validate(contents, schema=schema)
+
+
+def validate_hooks(repo_dir: Path) -> None:
+    """
+    Check that enable and disable hooks are present in the
+    repository, and that they have execute permissions.
+    """
+    for addon in get_addons_list(repo_dir):
+        addon_folder = repo_dir / "addons" / addon
+        for hook_name in ("enable", "disable"):
+            hook = addon_folder / hook_name
+            check_exists(hook)
+            check_is_executable(addon_folder)
+
+
+def get_addons_list(repo_dir: Path) -> List[str]:
+    contents = load_addons_yaml(repo_dir)
+    return [addon["name"] for addon in contents["microk8s-addons"]["addons"]]
```

Comment:
since we are validating the schema, are we sure that these do not crash at this point?